tests/data_engine/test_querying.py

```diff
@@ -27,6 +27,25 @@ def test_simple_filter(ds):
     assert q.to_dict() == expected
 
 
+def test_versioning(ds):
+    add_int_fields(ds, "x")
+    add_int_fields(ds, "y")
+
+    ds2 = ((ds[ds[Field("x", as_of_timestamp=123.99)] > 1]) &
+           (ds[ds[Field("x", as_of_timestamp=345)] > 2]) |
+           (ds[ds[Field("y", as_of_timestamp=789)] > 3])).\
+        select([Field("y", as_of_timestamp=123), Field("x", as_of_timestamp=456, alias="y_t1")])
+    q = ds2.get_query()
+    expected = {'or': {'children': [{'and': {'children': [{'gt': {'data': {'field': 'x', 'as_of': 123, 'value': 1}}},
+                                                          {'gt': {'data': {'field': 'x', 'as_of': 345, 'value': 2}}}],
+                                             'data': None}},
+                                    {'gt': {'data': {'field': 'y', 'as_of': 789, 'value': 3}}}], 'data': None}}
+    assert q.to_dict() == expected
+
+    expected_select = [{'name': 'y', 'asOf': 123}, {'name': 'x', 'asOf': 456, 'alias': 'y_t1'}]
+    assert ds2._select == expected_select
+
+
 def test_composite_filter(ds):
```

Comment:
After you implement the datetime handling, add tests here that ONLY handle the datetime stuff, don't overdo it with a complex composition query, just one select should be enough

Reply:
Also add a test for the dataset deserialization. Validate that if the recieved dict has a select field, that the select field shows up in the datasource.
It should be self evident, but I want it to be there just to be sure later. The serialization/deserialization stuff is changing a lot, so I want to be sure it has good test coverage

Reply:
1. not sure i understand, i added a test to the datetime/timestamp options its inside my existing test
2. done

Reply:
I want to have extremely basic true unit-tests that test one thing and one thing only.
A more complex test that tests interactions is also extremely valid, but if something changes in the code for handling e.g. specifically datetimes, the whole complex test fails, but it will still require combing to understand what exactly failed.
If we have a very barebones and simple test that specifically handles a timestamp in one case, and a datetime in another, any failing change would be quicker to triage